xapian-core/backends/glass/glass_database.cc

```diff
@@ -1081,7 +1081,8 @@ GlassWritableDatabase::add_document_(Xapian::docid did,
 
 		string tname = *term;
 		if (tname.size() > MAX_SAFE_TERM_LENGTH)
-		    throw Xapian::InvalidArgumentError("Term too long (> " STRINGIZE(MAX_SAFE_TERM_LENGTH) "): " + tname);
+		    tname.resize(MAX_SAFE_TERM_LENGTH);		
+		    //throw Xapian::InvalidArgumentError("Term too long (> " STRINGIZE(MAX_SAFE_TERM_LENGTH) "): " + tname);
```

Comment:
Instead of commenting out lines of code, you should just remove them; the version control history makes it easy to recover them if we need.